### ISSUE TITLE
Typescript support

### ISSUE THREAD

**Issue by ifiokjr** [jeffbski/wait-on#35]
@jeffbski I've just created a pull request (PR) for Typescript types on [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/32776). 

Would you prefer I opened the PR in this repository to colocate the types with the main codebase?

Instead of two packages `yarn add @types/wait-on wait-on`, users would only need to install one, `yarn add wait-on`.

**Comment by jeffbski** [comment]
yes, that would be great. Please do. Note, I have a fresh rewrite that I think is cleaner and easier to maintain. The external signatures should be pretty much the same.

**Comment by JessicaSachs** [comment]
> yes, that would be great. Please do. Note, I have a fresh rewrite that I think is cleaner and easier to maintain. The external signatures should be pretty much the same.

Just curious if this is still in the works 😄
